### ISSUE TITLE
Flutter desktop web support for Microsoft Edge Read Aloud

### ISSUE THREAD

**Issue by omatt** [flutter/flutter#121153]
Microsoft [Read Aloud](https://www.microsoft.com/en-us/edge/features/read-aloud) provides accessibility on reading through web pages. However, Read Aloud is unable to read text displayed on Flutter web apps.

## Steps to Reproduce

1.  Using Microsoft Edge, go to https://gallery.flutter.dev/#/reply
2. Click 'Read Aloud' available on the address bar. The 'Read Aloud' function can also be accessible from the menu. 
<img width="292" alt="Screenshot 2023-02-22 at 1 55 45 AM" src="https://user-images.githubusercontent.com/4143153/220422911-8a2c606a-7797-4a07-a349-71d26ffe89c9.png">

3. A toolbar for Read Aloud should appear and start reading the text available on the page.

**Expected results:** 

Read Aloud should be able to read text displayed in Flutter web on desktop.

**Actual results:** 

Read Aloud is unable to read the text displayed in Flutter web on desktop.

<details>
<summary>Code sample</summary>

```dart
```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I didn't find an existing issue related to this, so I am going ahead and labeling this for team's thoughts on feasibility and potential support.

**Comment by fideldonson** [comment]
I am seeing the same. Looks like the 'Read Aloud' extension has issues traversing the shadow dom.

**Comment by fideldonson** [comment]
This is a pretty serious issue from an accessibility stand point. Read aloud is integrated fully into edge - also on mobile platforms - and as such is the default option for tts for the users. Do you have any updates on ways to mitigate the issue.

**Comment by fideldonson** [comment]
I should say - the Read Aloud function in Edge has no problem reading/traversing shadow dom of other components utilising it - it seems there is a problem between flutter and Edge Read Aloud
